src/ws/session-utils.c

```diff
@@ -399,11 +399,61 @@ open_session (pam_handle_t *pamh)
   return PAM_SUCCESS;
 }
 
+static void
+do_fd_remapping (int *remap_fds, int n_remap_fds)
```

Comment:
I must say, I'm not a fan of this commit. We use this exacly once, with fixed arguments, so we don't need this complexity. It's much harder to validate and follow what this is doing than the two fixed dup2() calls. Is there a reason why this is necessary? The commit message should explain that then.